src/bin/gcc-shim.rs

```diff
@@ -1,48 +1,119 @@
 #![cfg_attr(test, allow(dead_code))]
 
 use std::env;
+use std::fmt;
 use std::fs::File;
-use std::io::prelude::*;
+use std::io::{self, prelude::*};
 use std::path::PathBuf;
 
+trait EnumUnwrapExt {
```

Comment:
I feel like this doesn't carry it's weight, if I'm being honest. Almost half the file is dedicated to this helper (56 lines out of 119, 47%), so I'd rather we just use `unwrap_or_else(|| panic!(...))` and such.

Especially given that the gcc-shim is mostly a test-only thing?

Reply:
I've replaced this with `unwrap_or_else` but I also include the `error` in `Result::unwrap_or_else`.